### PR TITLE
Per-user track changes

### PR DESCRIPTION
**Pull request by paulojreis**
We'll need to change https://github.com/sharelatex/web-sharelatex/pull/547/files#diff-7701684cedd84223921f73f784273c4eR239 to the date of the deploy just before going live

### CODE REVIEW

**Comment by jpallen** [comment]
At first, when turning on 'everyone', the toggles are selected. On refresh they are grey:

screen shot 2017-07-21 at 13 45 49

File: app/coffee/Features/Project/ProjectController.coffee [hunk 1]
```diff
@@ -230,14 +230,32 @@ module.exports = ProjectController =
 						return cb(null, false)
 					else
 						return cb(null, true)
+			showPerUserTCNotice: (cb) ->
+				cb = underscore.once(cb)
+				if !user_id?
+					return cb()
+				timestamp = user_id.toString().substring(0,8)
+				userSignupDate = new Date( parseInt( timestamp, 16 ) * 1000 )
+				if userSignupDate > new Date("2018-01-01") # 8th March
```
Comment (jpallen) [inline app/coffee/Features/Project/ProjectController.coffee]: Needs updating

File: public/coffee/ide/review-panel/controllers/ReviewPanelController.coffee [hunk 2]
```diff
@@ -565,24 +583,86 @@ define [
...
+		$scope.toggleTrackChangesForUser = (onForUser, userId) ->
+			_setUserTCState userId, onForUser, true
+			applyClientTrackChangesStateToServer()				
+
+		ide.socket.on "toggle-track-changes", (state) ->
 			$scope.$apply () ->
-				$scope.editor.wantTrackChanges = value
+				applyTrackChangesStateToClient(state)
+
+		# Not sure what the kbd shortcut should do now?
```
Comment (jpallen) [inline public/coffee/ide/review-panel/controllers/ReviewPanelController.coffee]: Did we have any solutions for this? Toggle it for the user maybe?

**Comment by paulojreis** [comment]
@jpallen , just addressed all your points:

* Toggles are "on" or "off" for all users, when the setting is applied to everyone, even on project load/refresh;
* Keyboard shortcut is re-enabled, toggles TC for the user;
* Sign-up date threshold to show the per-user TC prompt is updated.

Can you take a look again?

**Pull request merged**